### PR TITLE
Feature/chat screen

### PR DESCRIPTION
**Pull request by manema**
**### Chat Screen**

**Description:**
- Conversations retrieve.
- Chat Screen (display all user conversations)
- Chat box.

chat_menu

### CODE REVIEW

File: src/reducers/chatReducer.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Map } from 'immutable';
```
Comment (agustito37) [inline src/reducers/chatReducer.js]:
```suggestion
import { Map, fromJS } from 'immutable';
```

File: src/reducers/chatReducer.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+} from '../actions/actionTypes';
+
+export const initialState = Map({
+  conversations: []
+});
+
+const chatReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case FETCH_CONVERSATIONS_SUCCESS: {
+      return state.set('conversations', action.conversations);
```
Comment (agustito37) [inline src/reducers/chatReducer.js]:
```suggestion
      return state.set('conversations', fromJS(action.conversations));
```

File: src/selectors/chatSelector.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { createSelector } from 'reselect';
+
+export const obtainConversations = createSelector(
+  state => state.getIn(['chat', 'conversations']),
+  conversations => conversations
```
Comment (agustito37) [inline src/selectors/chatSelector.js]:
```suggestion
  conversations => conversations.toJS()
```

File: src/reducers/chatReducer.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Map } from 'immutable';
+import {
+  FETCH_CONVERSATIONS_SUCCESS
+} from '../actions/actionTypes';
+
+export const initialState = Map({
+  conversations: []
```
Comment (agustito37) [inline src/reducers/chatReducer.js]:
```suggestion
  conversations: List()
```

File: src/reducers/chatReducer.js (hunk 1 above)
Comment (manema) [reply src/reducers/chatReducer.js]: What a mistake :man-facepalming: , thank you for pointing it.

**Pull request merged**